examples/stest.py

```diff
@@ -361,6 +361,23 @@ def make_tests():
           "finest-u-error": 0.14767829457535478,
           "operator-complexity": 1.1666666666666667}]
 
+    tests["mlmc-sanity"] = \
+        [["./mlmc",
+          "--perm", spe10_perm_file],
+         {"finest-p-error": 0.10754186878360708}]
+
+    tests["mlmc-pde-sampler"] = \
+        [["./mlmc",
+          "--sampler-type", "pde",
+          "--kappa", "0.01"],
+         {"finest-p-error": 0.19865665579298755}]
```

Comment:
These tests (`mlmc-pde-sampler` and `par-mlmc`) are failing in my tux box. In `mlmc.cpp`, multiple samples are generated, which error will be checked? In fact, none of the samples match this error.

Reply:
Same here

Reply:
This test was intended to only check the last sample's error (that's where the python test scripts will look), which is not a very good test but at least tells you what I was intending.

Can you guys run this code snippet and tell me the result?
```C++
#include "../src/smoothG.hpp"

int main(int argc, char * argv[])
{
    smoothg::NormalDistribution nd0(0.0, 1.0, 0);
    std::cout << nd0.Sample() << " " << nd0.Sample() << " " << nd0.Sample() << std::endl;
    smoothg::NormalDistribution nd1(0.0, 1.0, 1);
    std::cout << nd1.Sample() << " " << nd1.Sample() << " " << nd1.Sample() << std::endl;
    return 0;
}
```
On my machine the result is
```
0.0708592 0.302805 1.12279
1.58275 -1.40287 -0.549746
```
Did we have a similar issue with the random graph generation? How did we make sure that was consistent across platforms?

Reply:
output on my machine:
```
./examples/testss 
1.12279 0.302805 0.0708592
-0.549746 -1.40287 1.58275
```
Transposed somehow?

Reply:
Another test:
``` c++
#include "../src/smoothG.hpp"

int main(int argc, char * argv[])
{
    smoothg::NormalDistribution nd0(0.0, 1.0, 0);
    double s1 = nd0.Sample();
    double s2 = nd0.Sample();
    double s3 = nd0.Sample();
    std::cout << s1 << " " << s2 << " " << s3 << std::endl;

    smoothg::NormalDistribution nd1(0.0, 1.0, 1);
    double s4 = nd1.Sample();
    double s5 = nd1.Sample();
    double s6 = nd1.Sample();
    std::cout << s4 << " " << s5 << " " << s6 << std::endl;

    return 0;
}
```
outputs:
```
1.12279 0.302805 0.0708592
-0.549746 -1.40287 1.58275
```

Reply:
When i run `mpirun -np 2 ./mlmc --sampler-type pde --kappa 0.01 --perm fake_perm.txt`, I get the same results as the test.  I'm guessing there is an issue reading the file and different permeabilities are being used.

If the user specified file is not found, that should actually throw/error (vs silently using `BlankPermeability`)

Reply:
Thanks @gelever , you're right, and I don't think it has anything to do with the random numbers. Hopefully fixed.

Reply:
@barker29 those tests now all pass on my tux box.